File: src/java/WebNinja/src/test/java/webninja/selenuim/IssueForm.java
package webninja.selenuim;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import webninja.User;
import webninja.UserRepository;

public class IssueForm {
    private final WebDriver webDriver;
    private final CodeTrack codeTrack;
    private final UserRepository userRepository;

    public IssueForm(WebDriver webDriver, CodeTrack codeTrack, UserRepository userRepository) {
        this.webDriver = webDriver;
        this.codeTrack = codeTrack;
        this.userRepository = userRepository;
    }

    public void setProjectName(String projectName) {
        codeTrack.selectOption(By.name("bug_data[Project]"), projectName);
    }

    public void setTitle(String title) {
        webDriver.findElement(By.name("bug_data[Summary]")).sendKeys(title);
    }

    public void setDescription(String description) {
        webDriver.findElement(By.name("bug_data[Description]")).sendKeys(description);
    }

    public void setSeverity(String severity) {
        codeTrack.selectOption(By.name("bug_data[Severity]"), severity);
    }

    public void assignTo(String userName) {
        User user = userRepository.getUserById(userName);
        String name = user == null ? userName : user.getName();
        codeTrack.selectOption(By.name("bug_data[Assign_To]"), name);
    }

}

File: src/dotnet/WebNinja/WebNinja/selenium/IssueForm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OpenQA.Selenium.Remote;
using WebNinja.webninja;

namespace WebNinja.selenium
{
    public class IssueForm : FormObject
    {

        public IssueForm(RemoteWebDriver driver, CodeTrack track, UserRepository repository) : base(driver)
        {
            throw new NotImplementedException();
        }

        public string ProjectName
        {
            set { throw new NotImplementedException(); }
        }

        public string Title
        {
            set { throw new NotImplementedException(); }
        }

        public string Description
        {
            set { throw new NotImplementedException(); }
        }

        public string Severity
        {
            set { throw new NotImplementedException(); }
        }

        public void AssignTo(string user)
        {
            throw new NotImplementedException();
        }
    }
}
